### DISCUSSION TITLE
Some basic questions about orientation fibers

### DISCUSSION THREAD

**Discussion by killuaLeo** [neperfepx/neper#918] [Q&A] [answered]
Hi All,

For the orientation fibers, there are two vectors defined, one is the crystal direction vector, another is the ideal fiber direction vector. I want to know if the textured case is the perfect fiber, how this rotation is made? I considered the rotation process should be rotating from crystal direction to the fiber direction at the first time. However, I have tried one case and the rotation angles I have got is not the same as what I have thought. Can anyone help me with this question?

Best,
Di

**Comment by mkasemer** [comment]
So we are on the same page: a crystallographic fiber is defined by:
1. A crystal direction
2. A sample direction

In other words, it defines the set of orientations which have a specific crystal direction aligned with a specific sample direction.

However... without further description of what, exactly, you think is incorrect we cannot properly diagnose your issue. Please provide a MWE, as well as code version you are using.

**Reply by killuaLeo** [reply]
Hi,

Thanks for your reply! I am trying to create a hcp polycrystal system with the crystal direction of (1, 0, 0) and the sample direction of (0.7071, 0.7071, 0). The input command shows as follows:
`neper -T -n 10 -domain "cube(0.01,0.01,0.001)" -morpho "columnar(z),gg" -crysym hexagonal -ori "fiber(1,0,0,0.7071,0.7071,0)" -o inclined10texturedplate2 -format tess -ori descriptor euler-bunge:active`
The rotation angles from (1,0,0) to (0.7071,0.7071,0) in Euler-Bunge convention should be 45, 0, 0. However the .tess file didn't show the same rotation angles. By the way, the version I'm using is 4.8.2-2.

The tesselation file output is shown below:
***tess
 **format
   3.4
 **general
   3 standard
 **cell
  10
  *crysym
   hexagonal
  *seed
   1 0.006573795799 0.004966463284 0.000500000000 0.002525151621
   2 0.003748635307 0.002189794547 0.000500000000 0.003211142648
   3 0.000837588783 0.009163960137 0.000500000000 0.000000000000
   4 0.003690521437 0.008786240411 0.000500000000 0.001913349507
   5 0.001296061509 0.006058105059 0.000500000000 0.000848029538
   6 0.007502271456 0.003567489836 0.000500000000 0.000000142653
   7 0.000503464383 0.001948616264 0.000500000000 0.002943262638
   8 0.008738556418 0.009563674974 0.000500000000 0.005820341864
   9 0.007888171659 0.004851784223 0.000500000000 0.001973236930
  10 0.005189821980 0.005909055911 0.000500000000 0.002493279442
  *ori
   euler-bunge:active
  45.000000000000   37.197179036517  -60.000000000000
-135.000000000000   77.644451093533  120.000000000000
-135.000000000000   76.639503389212  120.000000000000
-135.000000000000   74.478996824115  120.000000000000
-135.000000000000   49.348717797102  120.000000000000
-135.000000000000   71.980488627940  120.000000000000
  45.000000000000   88.126367666311  -60.000000000000
  45.000000000000   46.119024590915  -60.000000000000
  45.000000000000   19.202892350055  -60.000000000000
-135.000000000000   69.114813460200  120.000000000000
 **vertex
 44
   1  0.007974814530 0.002609749195 0.001000000000     0
   2  0.007974814530 0.002609749195 -0.000000000000     0
   3  0.005089661506 0.004359523842 -0.000000000000     0
   4  0.007269483994 0.002141632925 -0.000000000000     0
   5  0.007269483994 0.002141632925 0.001000000000     0
   6  0.005089661506 0.004359523842 0.001000000000     0
   7  0.008101725922 0.004064323974 0.001000000000     0
   8  0.005668852724 0.005209928841 0.001000000000     0
   9  0.005668852724 0.005209928841 -0.000000000000     0
  10  0.008101725922 0.004064323974 -0.000000000000     0
  11  0.002025816707 -0.000000000000 -0.000000000000     0
  12  0.002025816707 -0.000000000000 0.001000000000     0
  13  0.008055526507 -0.000000000000 -0.000000000000     0
  14  0.008055526507 -0.000000000000 0.001000000000     0
  15  0.002513082545 0.005357929411 0.001000000000     0
  16  0.001667465181 0.004821793808 0.001000000000     0
  17  0.001667465181 0.004821793808 -0.000000000000     0
  18  0.002513082545 0.005357929411 -0.000000000000     0
  19  0.000000000000 0.010000000000 0.000000000000     0
  20  0.000000000000 0.010000000000 0.001000000000     0
  21  0.001758144702 0.010000000000 0.001000000000     0
  22  0.001758144702 0.010000000000 -0.000000000000     0
  23  -0.000000000000 0.007569326472 -0.000000000000     0
  24  -0.000000000000 0.007569326472 0.001000000000     0
  25  0.001464961835 0.007785577722 0.001000000000     0
  26  0.001464961835 0.007785577722 -0.000000000000     0
  27  0.002568920602 0.006816643014 0.001000000000     0
  28  0.002568920602 0.006816643014 -0.000000000000     0
  29  0.003094680803 0.010000000000 -0.000000000000     0
  30  0.003094680803 0.010000000000 0.001000000000     0
  31  0.003509459913 0.007306757951 0.001000000000     0
  32  0.003509459913 0.007306757951 -0.000000000000     0
  33  -0.000000000000 0.005143397827 -0.000000000000     0
  34  -0.000000000000 0.005143397827 0.001000000000     0
  35  0.010000000000 0.000000000000 0.000000000000     0
  36  0.010000000000 0.000000000000 0.001000000000     0
  37  0.010000000000 0.002001228678 -0.000000000000     0
  38  0.010000000000 0.002001228678 0.001000000000     0
  39  0.000000000000 0.000000000000 0.000000000000     0
  40  0.000000000000 0.000000000000 0.001000000000     0
  41  0.010000000000 0.010000000000 0.000000000000     0
  42  0.010000000000 0.010000000000 0.001000000000     0
  43  0.010000000000 0.003721730431 0.001000000000     0
  44  0.010000000000 0.003721730431 -0.000000000000     0
 **edge
 84
   1  2 4 0
   2  4 3 0
   3  3 9 0
   4  9 10 0
   5  10 2 0
   6  6 5 0
   7  5 1 0
   8  1 7 0
   9  7 8 0
  10  8 6 0
  11  2 1 0
  12  7 10 0
  13  3 6 0
  14  8 9 0
  15  4 5 0
  16  12 11 0
  17  11 13 0
  18  13 14 0
  19  14 12 0
  20  4 13 0
  21  11 17 0
  22  17 18 0
  23  18 3 0
  24  14 5 0
  25  6 15 0
  26  15 16 0
  27  16 12 0
  28  16 17 0
  29  15 18 0
  30  20 19 0
  31  19 23 0
  32  23 24 0
  33  24 20 0
  34  20 21 0
  35  21 22 0
  36  22 19 0
  37  22 26 0
  38  26 23 0
  39  24 25 0
  40  25 21 0
  41  26 25 0
  42  21 30 0
  43  30 29 0
  44  29 22 0
  45  28 26 0
  46  29 32 0
  47  32 28 0
  48  25 27 0
  49  27 31 0
  50  31 30 0
  51  27 28 0
  52  32 31 0
  53  23 33 0
  54  33 34 0
  55  34 24 0
  56  28 18 0
  57  17 33 0
  58  34 16 0
  59  15 27 0
  60  36 35 0
  61  35 37 0
  62  37 38 0
  63  38 36 0
  64  36 14 0
  65  13 35 0
  66  2 37 0
  67  38 1 0
  68  39 40 0
  69  40 34 0
  70  33 39 0
  71  39 11 0
  72  12 40 0
  73  41 42 0
  74  42 43 0
  75  43 44 0
  76  44 41 0
  77  41 29 0
  78  30 42 0
  79  44 10 0
  80  9 32 0
  81  7 43 0
  82  31 8 0
  83  37 44 0
  84  43 38 0
 **face
 51
   1 5 2 4 3 9 10
     5 1 2 3 4 5
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   2 5 6 5 1 7 8
     5 6 7 8 9 10
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   3 4 2 1 7 10
     4 11 8 12 5
    -0.007717794092 -0.996215328717 0.086919611308 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   4 4 3 6 8 9
     4 13 -10 14 -3
    -0.001752611629 -0.826512396759 0.562918518086 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   5 4 1 2 4 5
     4 -11 1 15 7
    -0.002235467325 -0.552978262241 0.833195680191 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   6 4 3 4 5 6
     4 -2 15 -6 -13
    -0.006685801725 -0.713201069624 -0.700959509734 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   7 4 7 8 9 10
     4 9 14 4 -12
    -0.007128530029 -0.426017251163 -0.904715038955 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   8 4 12 11 13 14
     4 16 17 18 19
    -0.000000000000 -0.000000000000 -1.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
   9 6 3 4 13 11 17 18
     6 -2 20 -17 21 22 23
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  10 6 12 14 5 6 15 16
     6 -19 24 -6 25 26 27
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  11 4 11 12 16 17
     4 -16 -27 28 -21
    -0.002020245146 -0.997249721186 -0.074114732640 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  12 4 3 6 15 18
     4 13 25 29 23
    -0.005903982638 -0.361315159271 -0.932443754701 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  13 4 13 14 5 4
     4 18 24 -15 20
    -0.007562255761 -0.938766169309 -0.344554900358 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  14 4 15 16 17 18
     4 26 28 22 -29
    -0.003179417273 0.535463987428 -0.844558060862 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  15 4 20 19 23 24
     4 30 31 32 33
    -0.000000000000 -1.000000000000 -0.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  16 4 19 20 21 22
     4 -30 34 35 36
    0.010000000000 0.000000000000 1.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  17 4 19 22 26 23
     4 -36 37 38 -31
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  18 4 21 20 24 25
     4 -34 -33 39 40
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  19 4 21 22 26 25
     4 35 37 41 40
    -0.000430418689 -0.991349083655 0.131251645081 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  20 4 23 24 25 26
     4 32 39 -41 38
    0.007488181014 -0.146033136903 0.989279699037 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  21 4 22 21 30 29
     4 -35 42 43 44
    0.010000000000 0.000000000000 1.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  22 5 28 26 22 29 32
     5 45 -37 -44 46 47
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  23 5 21 25 27 31 30
     5 -40 48 49 50 -42
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  24 4 27 28 32 31
     4 51 -47 52 -49
    -0.004857965869 0.462120358594 -0.886817215762 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  25 4 25 27 28 26
     4 48 51 45 41
    0.006817795143 0.659649335909 0.751573518450 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  26 4 29 30 31 32
     4 -43 -50 -52 -46
    0.004580749082 0.988347739516 0.152212830579 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  27 4 24 23 33 34
     4 -32 53 54 55
    -0.000000000000 -1.000000000000 -0.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  28 6 33 23 26 28 18 17
     6 -53 -38 -45 56 -22 57
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  29 6 27 25 24 34 16 15
     6 -48 -39 -55 58 -26 59
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  30 4 27 28 18 15
     4 51 56 -29 59
    -0.002306297449 -0.999268164318 0.038250957905 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  31 4 33 34 16 17
     4 54 58 28 57
    0.005050322432 0.189379812326 0.981903909089 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  32 4 36 35 37 38
     4 60 61 62 63
    0.010000000000 1.000000000000 0.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  33 4 35 36 14 13
     4 -60 64 -18 65
    -0.000000000000 -0.000000000000 -1.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  34 5 2 37 35 13 4
     5 66 -61 -65 -20 -1
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  35 5 36 38 1 5 14
     5 -63 67 -7 -24 -64
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  36 4 37 38 1 2
     4 62 67 -11 66
    -0.004794242610 -0.287766468286 -0.957700610697 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  37 4 39 40 34 33
     4 68 69 -54 70
    -0.000000000000 -1.000000000000 -0.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  38 4 40 39 11 12
     4 -68 71 -16 72
    -0.000000000000 -0.000000000000 -1.000000000000 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  39 4 11 39 33 17
     4 -71 -70 -57 -21
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  40 4 40 12 16 34
     4 -72 -27 -58 -69
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  41 4 41 42 43 44
     4 73 74 75 76
    0.010000000000 1.000000000000 0.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  42 4 42 41 29 30
     4 -73 77 -43 78
    0.010000000000 0.000000000000 1.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  43 6 41 44 10 9 32 29
     6 -76 79 -4 80 -46 -77
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  44 6 8 7 43 42 30 31
     6 -9 81 -74 -78 -50 82
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  45 4 43 44 10 7
     4 75 79 -12 81
    0.005438630738 0.177607027223 0.984101490641 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  46 4 8 9 32 31
     4 14 80 52 82
    0.007686859747 0.696637512576 0.717423289329 -0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  47 4 38 37 44 43
     4 -62 83 -75 84
    0.010000000000 1.000000000000 0.000000000000 0.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  48 4 37 2 10 44
     4 -66 -5 -79 -83
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  49 4 1 38 43 7
     4 -67 -84 -81 -8
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  50 5 3 18 28 32 9
     5 -23 -56 -47 -80 -3
    -0.000000000000 -0.000000000000 -0.000000000000 -1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
  51 5 27 15 6 8 31
     5 -59 -25 -10 -82 -49
    0.001000000000 0.000000000000 0.000000000000 1.000000000000
     0 0 0.000000000000 0.000000000000 0.000000000000
 **polyhedron
 10
   1 7 1 2 -3 4 -5 6 -7
   2 8 8 9 10 11 -6 -12 -13 -14
   3 6 15 16 17 18 -19 -20
   4 7 21 22 23 19 24 -25 26
   5 8 27 28 29 20 25 -30 -31 14
   6 7 32 33 34 35 -36 5 13
   7 6 37 38 39 40 -11 31
   8 8 41 42 43 44 -45 7 -46 -26
   9 6 47 48 49 3 36 45
  10 7 50 51 -4 -24 30 12 46
 **domain
  *general
   cube
  *vertex
   8
   1  0.000000000000 0.000000000000 0.000000000000 x0y0z0
      1 39
   2  0.010000000000 0.000000000000 0.000000000000 x1y0z0
      1 35
   3  0.010000000000 0.010000000000 0.000000000000 x1y1z0
      1 41
   4  0.000000000000 0.010000000000 0.000000000000 x0y1z0
      1 19
   5  0.000000000000 0.000000000000 0.001000000000 x0y0z1
      1 40
   6  0.000000000000 0.010000000000 0.001000000000 x0y1z1
      1 20
   7  0.010000000000 0.010000000000 0.001000000000 x1y1z1
      1 42
   8  0.010000000000 0.000000000000 0.001000000000 x1y0z1
      1 36
  *edge
   12
   1 2 4 6
     x0y1
     1 30
   2 2 6 5
     x0z1
     3 33 55 69
   3 2 5 1
     x0y0
     1 68
   4 2 1 4
     x0z0
     3 31 53 70
   5 2 2 8
     x1y0
     1 60
   6 2 8 7
     x1z1
     3 63 74 84
   7 2 7 3
     x1y1
     1 73
   8 2 3 2
     x1z0
     3 61 76 83
   9 2 5 8
     y0z1
     3 19 64 72
  10 2 2 1
     y0z0
     3 17 65 71
  11 2 4 3
     y1z0
     3 36 44 77
  12 2 7 6
     y1z1
     3 34 42 78
  *face
   6
   1 4 4 6 5 1
     4 1 2 3 4
     plane
     4 0.000000000000 1.000000000000 0.000000000000 0.000000000000
     x0
     3 15 27 37
   2 4 2 8 7 3
     4 5 6 7 8
     plane
     4 -0.010000000000 -1.000000000000 -0.000000000000 -0.000000000000
     x1
     3 32 41 47
   3 4 1 5 8 2
     4 3 9 5 10
     plane
     4 0.000000000000 0.000000000000 1.000000000000 0.000000000000
     y0
     3 8 33 38
   4 4 4 3 7 6
     4 11 7 12 1
     plane
     4 -0.010000000000 -0.000000000000 -1.000000000000 -0.000000000000
     y1
     3 16 21 42
   5 4 1 2 3 4
     4 10 8 11 4
     plane
     4 0.000000000000 0.000000000000 0.000000000000 1.000000000000
     z0
    10 1 9 17 22 28 34 39 43 48 50
   6 4 5 6 7 8
     4 2 12 6 9
     plane
     4 -0.001000000000 -0.000000000000 -0.000000000000 -1.000000000000
     z1
    10 2 10 18 23 29 35 40 44 49 51
***end
__

**Reply by mkasemer** [reply]
I would suggest taking some of those Euler-Bunge angles, converting to a rotation matrix, and transforming a `001` vector to the sample frame. This is the test to make sure those orientations yield something as-expected. Is this the case (my suspicion is yes, but I will let you do the verification).

**Reply by killuaLeo** [reply]
For the first angle set, the rotation matrix is 
```
[0.8413   -0.1342   -0.5236
    0.3307    0.8940    0.3023
    0.4275   -0.4275    0.7966]
```
Using this to rotate `001` vector will get `-0.5236, 0.3023, 0.7966` vector, also using this to rotate `100` vector will get `0.8413, 0.3307, 0.4275` vector.

For the second angle set, the rotation matrix is
```
[0.4846    0.2225    0.8460
    0.5367    0.6880   -0.4884
   -0.6907    0.6907    0.2140]
```
Using this to rotate `001` vector will get `0.8460, -0.4884, 0.2140` vector, also using this to rotate `100` vector will get `0.4846, 0.5367, -0.6907` vector.

**Reply by rquey** [reply, marked as answer]
@killuaLeo, you said:

> The rotation angles from (1,0,0) to (0.7071,0.7071,0) in Euler-Bunge convention should be 45, 0, 0. However the .tess file didn't show the same rotation angles.

and the orientations are:
```
*ori
euler-bunge:active
45.000000000000 37.197179036517 -60.000000000000
-135.000000000000 77.644451093533 120.000000000000
-135.000000000000 76.639503389212 120.000000000000
-135.000000000000 74.478996824115 120.000000000000
-135.000000000000 49.348717797102 120.000000000000
-135.000000000000 71.980488627940 120.000000000000
45.000000000000 88.126367666311 -60.000000000000
45.000000000000 46.119024590915 -60.000000000000
45.000000000000 19.202892350055 -60.000000000000
-135.000000000000 69.114813460200 120.000000000000
```

(1, 0, 0) | (0.707, 0.707, 0) is indeed obtained by a rotation of 45 degrees about Z, and this is why you get either 45 or (the equivalent -135) as first Euler angle. But  a fiber is not just one orientation. It is all orientations obtained by rotation about the axis of the fiber, i.e. the 45°-tilted x. This is actually the second Euler angle. As for the third one, -60 and 120 is equivalent to 0 under hexagonal crystal symmetry.

In conclusion, the orientations do describe the specified orientation fiber.

PS. You should update to 4.10.1.

**Reply by mkasemer** [reply]
@killuaLeo I was not aware that you were working with hexagonal symmetry, as I did not read closely enough. I also had a question on this, but @rquey is indeed correct.

Note that Neper will generate any orientation that lies along that fiber, and for any of the symmetrically equivalent crystallographic directions. For your case, you are interested in `1 0 0`. A symmetrically equivalent crystallographic direction, would be (in MATLAB):
```
>> c = [sind(30) cosd(30) 0]';
```

When using the rotation matrix (which you have transposed, as @rquey said, please update Neper---**EDIT**: I may be incorrectly transposing, so please ignore that.):
```
>> R*c
ans =
    0.7071
    0.7071
    0.0000
```

Hope this helps clear up.
